news-bundle/src/Security/Voter/NewsArchiveAccessVoter.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\NewsBundle\Security\Voter;
+
+use Contao\CoreBundle\Security\ContaoCorePermissions;
+use Contao\CoreBundle\Security\DataContainer\DataContainerSubject;
+use Contao\NewsBundle\Security\ContaoNewsPermissions;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+use Symfony\Component\Security\Core\Security;
+
+class NewsArchiveAccessVoter extends Voter
+{
+    public function __construct(private Security $security)
+    {
+    }
+
+    protected function supports(string $attribute, $subject)
+    {
+        return $subject instanceof DataContainerSubject &&
+            'tl_news_archive' === $subject->table &&
+            str_starts_with($attribute, ContaoCorePermissions::DC_ACTION_PREFIX);
+    }
+
+    protected function voteOnAttribute(string $attribute, $subject, TokenInterface $token)
+    {
+        return match ($attribute) {
+            ContaoCorePermissions::DC_ACTION_CREATE => $this->security->isGranted(ContaoNewsPermissions::USER_CAN_CREATE_ARCHIVES),
+            ContaoCorePermissions::DC_ACTION_COPY => $this->security->isGranted(ContaoNewsPermissions::USER_CAN_CREATE_ARCHIVES) &&
+                $this->security->isGranted(ContaoNewsPermissions::USER_CAN_EDIT_ARCHIVE, $subject->id),
+            ContaoCorePermissions::DC_ACTION_EDIT,
```

Comment:
This should also check for `USER_CAN_EDIT_FIELDS_OF_TABLE` shouldn't it?

Reply:
No, I don't think this is correct. This should be a general voter because it's always the same. You are never allowed to edit if you cannot edit any field of a table. That's always true for all tables.
I wonder if this should be checked in the `DC_` itself even, wdyt?

Reply:
yeah, it even applies to CREATE or COPY permissions, since you would not be able to edit the record afterwards. I think having a generic voter for this would be an awesome improvement since that currently is not correctly checked. But that should be a separate PR 😊